client/main-layout.tsx

```diff
@@ -0,0 +1,401 @@
+    />
+  ) : (
+    <ActivityButton
+      key='searching-match'
+      ref={searchingMatchButtonRef}
+      icon={<FindMatchIcon />}
+      glowing={true}
+      label='Finding…'
+      onClick={() => setSearchingMatchOverlayOpen(true)}
+      hotkey={{ keyCode: KEY_F, altKey: true }}
+    />
+  )
+  const activityButtons = IS_ELECTRON
+    ? [
+        findMatchButton,
+        <ActivityButton
+          key='create-game'
+          icon={<CreateGameIcon />}
+          label='Create'
+          onClick={onCreateLobbyClick}
+          disabled={inGameplayActivity}
+          hotkey={{ keyCode: KEY_C, altKey: true }}
+        />,
+        <ActivityButton
+          key='join-game'
+          icon={<JoinGameIcon />}
+          label='Join'
+          onClick={onJoinLobbyClick}
+          hotkey={{ keyCode: KEY_J, altKey: true }}
+          count={lobbyCount > 0 ? lobbyCount : undefined}
+        />,
+        <ActivityButton
+          key='maps'
+          icon={<StyledMapsIcon />}
+          label='Maps'
+          onClick={onMapsClick}
+          hotkey={{ keyCode: KEY_M, altKey: true }}
+        />,
+        <ActivityButton
+          key='replays'
+          icon={<ReplaysIcon />}
+          label='Replays'
+          onClick={onReplaysClick}
+          hotkey={{ keyCode: KEY_R, altKey: true }}
+        />,
+        <ActivityButton
+          key='ladder'
+          icon={<LadderIcon />}
+          label='Ladder'
+          onClick={() => navigateToLadder()}
+          hotkey={{ keyCode: KEY_D, altKey: true }}
+        />,
+        <ActivitySpacer key='spacer' />,
+      ]
+    : [
+        <ActivityButton
+          key='download'
+          icon={<DownloadIcon />}
+          label='Download'
+          onClick={() => dispatch(openDialog(DialogType.Download))}
+        />,
+        <ActivityButton
+          key='ladder'
+          icon={<LadderIcon />}
+          label='Ladder'
+          onClick={() => navigateToLadder()}
+          hotkey={{ keyCode: KEY_D, altKey: true }}
+        />,
+        <ActivitySpacer key='spacer' />,
+      ]
+
+  const renderSearchingMatchOverlay = () => {
+    if (!IS_ELECTRON && !isMatchmakingSearching) {
+      return null
+    }
+
+    return (
+      <MatchmakingSearchingOverlay
+        open={searchingMatchOverlayOpen}
+        anchor={searchingMatchButtonRef.current}
+        onCancelSearch={() => {
+          dispatch(cancelFindMatch())
+          setSearchingMatchOverlayOpen(false)
+        }}
+        onDismiss={() => setSearchingMatchOverlayOpen(false)}
+      />
+    )
+  }
+
+  return (
+    <Container>
+      <ConnectedLeftNav />
+      <Content>
+        <Switch>
+          <ConditionalRoute
+            path='/admin/:rest*'
+            filters={[IsAdminFilter]}
+            component={LoadableAdminPanel}
+          />
+          <Route path='/chat' component={ChatList} />
+          <Route path='/chat/:channel' component={ChatChannel} />
+          <Route path='/ladder/:matchmakingLabel?'>
```

Comment:
1) Why is this "matchmakingLabel" and not "matchmakingType"?
2) imo this isn't a great approach for these because it makes it hard for these components to add or change the parameters they accept. Ideally they would be the maintainers of all routes underneath their respective root path, and not have to come back up and modify main-layout every time they make a change (thus, this should simply be `/ladder/:rest*` and pass all the parameters into a component in the ladder package). Apply throughout this file